### ISSUE TITLE
Use tf.get_variable() instead of tf.Variable, parameter did not update in every epoch?

### ISSUE THREAD

**Issue by fengsky401** [tensorflow/tensorflow#6890]
I use the program in the page 
https://github.com/tensorflow/tensorflow/tree/master/tensorflow/examples/tutorials/mnist
, just make tf.get_variable() instead of tf.Variable(), other parts are all the same, the result is very different, I doubt whether the parameter update in every epoch? 
# The result of using tf.get_variable():
Step 0: loss = 449.45 (0.013 sec)
Step 100: loss = 10.57 (0.004 sec)
Step 200: loss = 3.80 (0.005 sec)
Step 300: loss = 2.06 (0.004 sec)
Step 400: loss = 1.99 (0.004 sec)
Step 500: loss = 2.29 (0.004 sec)
Step 600: loss = 1.77 (0.005 sec)
Step 700: loss = 2.01 (0.004 sec)
Step 800: loss = 2.39 (0.007 sec)
Step 900: loss = 1.71 (0.004 sec)
Training Data Eval:
  Num examples: 55000  Num correct: 21297  Precision @ 1: 0.3872
Validation Data Eval:
  Num examples: 5000  Num correct: 1886  Precision @ 1: 0.3772
Test Data Eval:
  Num examples: 10000  Num correct: 3780  Precision @ 1: 0.3780
Step 1000: loss = 2.27 (0.009 sec)
Step 1100: loss = 2.45 (0.073 sec)
Step 1200: loss = 1.57 (0.004 sec)
Step 1300: loss = 1.80 (0.004 sec)
Step 1400: loss = 1.49 (0.004 sec)
Step 1500: loss = 1.84 (0.004 sec)
Step 1600: loss = 1.34 (0.004 sec)
Step 1700: loss = 1.44 (0.004 sec)
Step 1800: loss = 1.56 (0.004 sec)
Step 1900: loss = 1.48 (0.004 sec)
Training Data Eval:
  Num examples: 55000  Num correct: 26545  Precision @ 1: 0.4826
Validation Data Eval:
  Num examples: 5000  Num correct: 2336  Precision @ 1: 0.4672
Test Data Eval:
  Num examples: 10000  Num correct: 4785  Precision @ 1: 0.4785
Step 2000: loss = 1.58 (0.011 sec)
Step 2100: loss = 1.37 (0.004 sec)
Step 2200: loss = 1.50 (0.070 sec)
Step 2300: loss = 1.38 (0.004 sec)
Step 2400: loss = 1.72 (0.004 sec)
Step 2500: loss = 1.58 (0.004 sec)
Step 2600: loss = 1.20 (0.004 sec)
Step 2700: loss = 1.34 (0.004 sec)
Step 2800: loss = 1.49 (0.004 sec)
Step 2900: loss = 1.15 (0.004 sec)
Training Data Eval:
  Num examples: 55000  Num correct: 30216  Precision @ 1: 0.5494
Validation Data Eval:
  Num examples: 5000  Num correct: 2685  Precision @ 1: 0.5370
Test Data Eval:
  Num examples: 10000  Num correct: 5481  Precision @ 1: 0.5481
Step 3000: loss = 1.19 (0.010 sec)
Step 3100: loss = 1.69 (0.004 sec)
Step 3200: loss = 1.25 (0.004 sec)
Step 3300: loss = 1.25 (0.071 sec)
Step 3400: loss = 1.14 (0.004 sec)
Step 3500: loss = 1.27 (0.004 sec)
Step 3600: loss = 1.50 (0.004 sec)
Step 3700: loss = 1.07 (0.004 sec)
Step 3800: loss = 1.15 (0.004 sec)
Step 3900: loss = 1.12 (0.004 sec)
Training Data Eval:
  Num examples: 55000  Num correct: 29304  Precision @ 1: 0.5328
Validation Data Eval:
  Num examples: 5000  Num correct: 2684  Precision @ 1: 0.5368
Test Data Eval:
  Num examples: 10000  Num correct: 5234  Precision @ 1: 0.5234
Step 4000: loss = 1.13 (0.008 sec)
Step 4100: loss = 1.23 (0.004 sec)
Step 4200: loss = 1.24 (0.004 sec)
Step 4300: loss = 1.20 (0.004 sec)
Step 4400: loss = 1.03 (0.069 sec)
Step 4500: loss = 0.89 (0.004 sec)
Step 4600: loss = 1.37 (0.004 sec)
Step 4700: loss = 1.13 (0.004 sec)
Step 4800: loss = 1.17 (0.004 sec)
Step 4900: loss = 1.29 (0.004 sec)
Training Data Eval:
  Num examples: 55000  Num correct: 32965  Precision @ 1: 0.5994
Validation Data Eval:
  Num examples: 5000  Num correct: 2963  Precision @ 1: 0.5926
Test Data Eval:
  Num examples: 10000  Num correct: 5944  Precision @ 1: 0.5944
Step 5000: loss = 1.20 (0.008 sec)
Step 5100: loss = 1.28 (0.004 sec)
Step 5200: loss = 1.09 (0.004 sec)
Step 5300: loss = 0.94 (0.004 sec)
Step 5400: loss = 1.09 (0.004 sec)
Step 5500: loss = 1.23 (0.069 sec)
Step 5600: loss = 1.03 (0.004 sec)
Step 5700: loss = 1.14 (0.004 sec)
Step 5800: loss = 1.25 (0.004 sec)
Step 5900: loss = 1.28 (0.004 sec)
Training Data Eval:
  Num examples: 55000  Num correct: 33732  Precision @ 1: 0.6133
Validation Data Eval:
  Num examples: 5000  Num correct: 3071  Precision @ 1: 0.6142
Test Data Eval:
  Num examples: 10000  Num correct: 6170  Precision @ 1: 0.6170
Step 6000: loss = 0.99 (0.007 sec)
Step 6100: loss = 1.09 (0.004 sec)
Step 6200: loss = 1.00 (0.004 sec)
Step 6300: loss = 1.00 (0.004 sec)
Step 6400: loss = 1.28 (0.004 sec)
Step 6500: loss = 0.90 (0.004 sec)
Step 6600: loss = 0.73 (0.069 sec)
Step 6700: loss = 1.08 (0.004 sec)
Step 6800: loss = 0.99 (0.004 sec)
Step 6900: loss = 0.89 (0.004 sec)
Training Data Eval:
  Num examples: 55000  Num correct: 32685  Precision @ 1: 0.5943
Validation Data Eval:
  Num examples: 5000  Num correct: 2972  Precision @ 1: 0.5944
Test Data Eval:
  Num examples: 10000  Num correct: 5963  Precision @ 1: 0.5963
Step 7000: loss = 1.10 (0.009 sec)
Step 7100: loss = 1.32 (0.004 sec)
Step 7200: loss = 0.71 (0.004 sec)
Step 7300: loss = 0.88 (0.004 sec)
Step 7400: loss = 1.04 (0.004 sec)
Step 7500: loss = 1.69 (0.004 sec)
Step 7600: loss = 1.03 (0.004 sec)
Step 7700: loss = 1.32 (0.069 sec)
Step 7800: loss = 0.98 (0.004 sec)
Step 7900: loss = 1.31 (0.004 sec)
Training Data Eval:
  Num examples: 55000  Num correct: 36517  Precision @ 1: 0.6639
Validation Data Eval:
  Num examples: 5000  Num correct: 3351  Precision @ 1: 0.6702
Test Data Eval:
  Num examples: 10000  Num correct: 6659  Precision @ 1: 0.6659
Step 8000: loss = 0.91 (0.009 sec)
Step 8100: loss = 0.91 (0.004 sec)
Step 8200: loss = 0.95 (0.004 sec)
Step 8300: loss = 1.14 (0.004 sec)
Step 8400: loss = 1.00 (0.004 sec)
Step 8500: loss = 0.97 (0.004 sec)
Step 8600: loss = 0.73 (0.004 sec)
Step 8700: loss = 1.02 (0.004 sec)
Step 8800: loss = 0.81 (0.069 sec)
Step 8900: loss = 0.85 (0.004 sec)
Training Data Eval:
  Num examples: 55000  Num correct: 37801  Precision @ 1: 0.6873
Validation Data Eval:
  Num examples: 5000  Num correct: 3456  Precision @ 1: 0.6912
Test Data Eval:
  Num examples: 10000  Num correct: 6916  Precision @ 1: 0.6916
Step 9000: loss = 0.59 (0.010 sec)
Step 9100: loss = 1.02 (0.004 sec)
Step 9200: loss = 0.84 (0.004 sec)
Step 9300: loss = 0.79 (0.004 sec)
Step 9400: loss = 0.88 (0.004 sec)
Step 9500: loss = 0.94 (0.004 sec)
Step 9600: loss = 1.05 (0.004 sec)
Step 9700: loss = 0.78 (0.004 sec)
Step 9800: loss = 1.08 (0.004 sec)
Step 9900: loss = 1.12 (0.070 sec)
Training Data Eval:
  Num examples: 55000  Num correct: 38346  Precision @ 1: 0.6972
Validation Data Eval:
  Num examples: 5000  Num correct: 3526  Precision @ 1: 0.7052
Test Data Eval:
  Num examples: 10000  Num correct: 6997  Precision @ 1: 0.6997
[Finished in 60.3s]

# The result of using tf.Variable():
Extracting Mnist_data/train-images-idx3-ubyte.gz
Extracting Mnist_data/train-labels-idx1-ubyte.gz
Extracting Mnist_data/t10k-images-idx3-ubyte.gz
Extracting Mnist_data/t10k-labels-idx1-ubyte.gz
WARNING:tensorflow:Passing a `GraphDef` to the SummaryWriter is deprecated. Pass a `Graph` object instead, such as `sess.graph`.
Step 0: loss = 2.33 (0.013 sec)
Step 100: loss = 2.16 (0.004 sec)
Step 200: loss = 1.90 (0.004 sec)
Step 300: loss = 1.56 (0.004 sec)
Step 400: loss = 1.27 (0.004 sec)
Step 500: loss = 0.84 (0.006 sec)
Step 600: loss = 0.83 (0.004 sec)
Step 700: loss = 0.82 (0.004 sec)
Step 800: loss = 0.56 (0.004 sec)
Step 900: loss = 0.50 (0.005 sec)
Training Data Eval:
  Num examples: 55000  Num correct: 47541  Precision @ 1: 0.8644
Validation Data Eval:
  Num examples: 5000  Num correct: 4351  Precision @ 1: 0.8702
Test Data Eval:
  Num examples: 10000  Num correct: 8708  Precision @ 1: 0.8708
Step 1000: loss = 0.56 (0.009 sec)
Step 1100: loss = 0.49 (0.090 sec)
Step 1200: loss = 0.63 (0.004 sec)
Step 1300: loss = 0.38 (0.004 sec)
Step 1400: loss = 0.35 (0.004 sec)
Step 1500: loss = 0.46 (0.004 sec)
Step 1600: loss = 0.40 (0.006 sec)
Step 1700: loss = 0.47 (0.004 sec)
Step 1800: loss = 0.33 (0.004 sec)
Step 1900: loss = 0.30 (0.004 sec)
Training Data Eval:
  Num examples: 55000  Num correct: 49230  Precision @ 1: 0.8951
Validation Data Eval:
  Num examples: 5000  Num correct: 4511  Precision @ 1: 0.9022
Test Data Eval:
  Num examples: 10000  Num correct: 8998  Precision @ 1: 0.8998
Step 2000: loss = 0.33 (0.009 sec)
Step 2100: loss = 0.33 (0.004 sec)
Step 2200: loss = 0.34 (0.069 sec)
Step 2300: loss = 0.45 (0.004 sec)
Step 2400: loss = 0.31 (0.004 sec)
Step 2500: loss = 0.29 (0.005 sec)
Step 2600: loss = 0.17 (0.004 sec)
Step 2700: loss = 0.23 (0.004 sec)
Step 2800: loss = 0.34 (0.006 sec)
Step 2900: loss = 0.39 (0.004 sec)
Training Data Eval:
  Num examples: 55000  Num correct: 49923  Precision @ 1: 0.9077
Validation Data Eval:
  Num examples: 5000  Num correct: 4585  Precision @ 1: 0.9170
Test Data Eval:
  Num examples: 10000  Num correct: 9114  Precision @ 1: 0.9114
Step 3000: loss = 0.18 (0.007 sec)
Step 3100: loss = 0.25 (0.004 sec)
Step 3200: loss = 0.41 (0.004 sec)
Step 3300: loss = 0.67 (0.071 sec)
Step 3400: loss = 0.28 (0.004 sec)
Step 3500: loss = 0.27 (0.004 sec)
Step 3600: loss = 0.34 (0.004 sec)
Step 3700: loss = 0.50 (0.004 sec)
Step 3800: loss = 0.27 (0.005 sec)
Step 3900: loss = 0.54 (0.004 sec)
Training Data Eval:
  Num examples: 55000  Num correct: 50419  Precision @ 1: 0.9167
Validation Data Eval:
  Num examples: 5000  Num correct: 4627  Precision @ 1: 0.9254
Test Data Eval:
  Num examples: 10000  Num correct: 9205  Precision @ 1: 0.9205
Step 4000: loss = 0.43 (0.009 sec)
Step 4100: loss = 0.31 (0.004 sec)
Step 4200: loss = 0.31 (0.005 sec)
Step 4300: loss = 0.20 (0.004 sec)
Step 4400: loss = 0.24 (0.070 sec)
Step 4500: loss = 0.40 (0.005 sec)
Step 4600: loss = 0.27 (0.004 sec)
Step 4700: loss = 0.32 (0.005 sec)
Step 4800: loss = 0.30 (0.004 sec)
Step 4900: loss = 0.24 (0.004 sec)
Training Data Eval:
  Num examples: 55000  Num correct: 50727  Precision @ 1: 0.9223
Validation Data Eval:
  Num examples: 5000  Num correct: 4656  Precision @ 1: 0.9312
Test Data Eval:
  Num examples: 10000  Num correct: 9284  Precision @ 1: 0.9284
Step 5000: loss = 0.40 (0.010 sec)
Step 5100: loss = 0.22 (0.004 sec)
Step 5200: loss = 0.29 (0.004 sec)
Step 5300: loss = 0.21 (0.004 sec)
Step 5400: loss = 0.29 (0.004 sec)
Step 5500: loss = 0.27 (0.072 sec)
Step 5600: loss = 0.17 (0.004 sec)
Step 5700: loss = 0.14 (0.004 sec)
Step 5800: loss = 0.21 (0.004 sec)
Step 5900: loss = 0.32 (0.004 sec)
Training Data Eval:
  Num examples: 55000  Num correct: 51058  Precision @ 1: 0.9283
Validation Data Eval:
  Num examples: 5000  Num correct: 4677  Precision @ 1: 0.9354
Test Data Eval:
  Num examples: 10000  Num correct: 9329  Precision @ 1: 0.9329
Step 6000: loss = 0.17 (0.010 sec)
Step 6100: loss = 0.35 (0.004 sec)
Step 6200: loss = 0.21 (0.004 sec)
Step 6300: loss = 0.23 (0.004 sec)
Step 6400: loss = 0.15 (0.004 sec)
Step 6500: loss = 0.34 (0.004 sec)
Step 6600: loss = 0.17 (0.074 sec)
Step 6700: loss = 0.29 (0.004 sec)
Step 6800: loss = 0.20 (0.004 sec)
Step 6900: loss = 0.21 (0.004 sec)
Training Data Eval:
  Num examples: 55000  Num correct: 51370  Precision @ 1: 0.9340
Validation Data Eval:
  Num examples: 5000  Num correct: 4697  Precision @ 1: 0.9394
Test Data Eval:
  Num examples: 10000  Num correct: 9387  Precision @ 1: 0.9387
Step 7000: loss = 0.23 (0.007 sec)
Step 7100: loss = 0.25 (0.004 sec)
Step 7200: loss = 0.16 (0.004 sec)
Step 7300: loss = 0.22 (0.004 sec)
Step 7400: loss = 0.12 (0.004 sec)
Step 7500: loss = 0.31 (0.004 sec)
Step 7600: loss = 0.21 (0.004 sec)
Step 7700: loss = 0.16 (0.071 sec)
Step 7800: loss = 0.18 (0.004 sec)
Step 7900: loss = 0.21 (0.004 sec)
Training Data Eval:
  Num examples: 55000  Num correct: 51623  Precision @ 1: 0.9386
Validation Data Eval:
  Num examples: 5000  Num correct: 4725  Precision @ 1: 0.9450
Test Data Eval:
  Num examples: 10000  Num correct: 9409  Precision @ 1: 0.9409
Step 8000: loss = 0.26 (0.008 sec)
Step 8100: loss = 0.23 (0.004 sec)
Step 8200: loss = 0.32 (0.004 sec)
Step 8300: loss = 0.16 (0.004 sec)
Step 8400: loss = 0.18 (0.004 sec)
Step 8500: loss = 0.21 (0.004 sec)
Step 8600: loss = 0.26 (0.004 sec)
Step 8700: loss = 0.17 (0.004 sec)
Step 8800: loss = 0.24 (0.070 sec)
Step 8900: loss = 0.12 (0.004 sec)
Training Data Eval:
  Num examples: 55000  Num correct: 51853  Precision @ 1: 0.9428
Validation Data Eval:
  Num examples: 5000  Num correct: 4739  Precision @ 1: 0.9478
Test Data Eval:
  Num examples: 10000  Num correct: 9444  Precision @ 1: 0.9444
Step 9000: loss = 0.26 (0.009 sec)
Step 9100: loss = 0.13 (0.004 sec)
Step 9200: loss = 0.18 (0.004 sec)
Step 9300: loss = 0.21 (0.004 sec)
Step 9400: loss = 0.13 (0.004 sec)
Step 9500: loss = 0.17 (0.004 sec)
Step 9600: loss = 0.22 (0.004 sec)
Step 9700: loss = 0.10 (0.004 sec)
Step 9800: loss = 0.23 (0.004 sec)
Step 9900: loss = 0.14 (0.073 sec)
Training Data Eval:
  Num examples: 55000  Num correct: 52047  Precision @ 1: 0.9463
Validation Data Eval:
  Num examples: 5000  Num correct: 4757  Precision @ 1: 0.9514
Test Data Eval:
  Num examples: 10000  Num correct: 9481  Precision @ 1: 0.9481
[Finished in 61.5s]
